### DISCUSSION TITLE
How _exactly_ is a modify.dkim block defined in maddy.conf?

### DISCUSSION THREAD

**Discussion by mcgrue** [emersion/maddy#635] [General]
I've tried several variations on `modify.dkim {...}` and `modify { dkim ... }` as per the docs at https://github.com/foxcpp/maddy/blob/master/docs/reference/modifiers/dkim.md and I keep getting:

`unknown module or global directive: modify`
or
`Unused configuration block at /etc/maddy/maddy.conf:186 - modify.dkim (modify.dkim)`

What am I doing wrong here?  (I just want to set `newkey_algo ed25519`.  the default key is too big for Route53's TXT records ;_;)

**Comment by mcgrue** [comment]
I see.  It's:

```
submission {
    modify {
        dkim { ...
```

Am I wrong that  https://github.com/foxcpp/maddy/blob/master/docs/reference/modifiers/dkim.md isn't referenced on the website, though?  I had to discover it by searching through the code on github.

**Reply by foxcpp** [reply]
Here it is: https://maddy.email/reference/modifiers/dkim/, named "DKIM signing" in sidebar.
